workflows/rnaseq/main.nf

```diff
@@ -866,11 +870,31 @@ workflow RNASEQ {
         ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))
         ch_multiqc_files = ch_multiqc_files.mix(ch_collated_versions)
 
+        // Derive a set of rename patterns such that any sample names derived
+        // from input fastqs in MultiQC are replaced with specified sample name
+        // strings
+
+        ch_name_replacements = ch_samples
+            .map{
+                def read1 = it[1].name.split('\\.')[0] + "\t" + it[0].id + '_1'
```

Comment:
I have now spent half an hour inspecting the code of [FilePathValidator of the nf-validation plugin](https://github.com/nextflow-io/nf-validation/blob/0edaedfd172d8d26f539b0f33a4443d3c0ac789c/plugins/nf-validation/src/main/nextflow/validation/FormatValidators/FilePathValidator.groovy#L4) and the [File Class of Nextflow](https://github.com/nextflow-io/nextflow/blob/3c65ceb13fbf49bb4d8915457a004538d9a5b485/modules/nf-commons/src/main/nextflow/file/FileHelper.groovy) trying to understand where the ` .name` property comes from and what other property or method might be available that could right away return you the cleaned basename without extension, but to no avail. 

I just trust now that this will work, because I simply don't know better.

Reply:
`simpleName` is what you were looking for, and yes, I should have been using that - done now.

Reply:
How does one find such information? Is there are function like `.view()` or a reference documentation for the classes?

Reply:
It's here: https://www.nextflow.io/docs/latest/script.html#getting-file-attributes

(Note the tip on accessing as a field)